Remotion/Web/Core/UI/SmartPage.cs

```diff
@@ -669,6 +669,13 @@ protected virtual bool IsCspReportOnlyEnabled
 
   protected override void OnInit (EventArgs e)
   {
+    if (AsyncMode)
```

Comment:
We don't need a second check in oOnLoadComplete or OnPreRender, correct? i.e. it's not possible to enable AsyncMode after OnInit anyway?

Reply:
The `AsyncMode` property itself does nothing. `Async="true"` affects the generated class and will set `AsyncMode = true;` in the constructor. So it is a good way to detect `Async="true"` but there is no reason why `AsyncMode` should be set for any other reason. So yes we are fine checking just here in `OnInit` :)

Reply:
:)